app/javascript/components/rooms/RoomJoin.jsx

```diff
@@ -60,10 +66,18 @@ export default function RoomJoin() {
               </div>
             ) : (
               <>
-                { room?.viewer_access_code
+                <label htmlFor="join-name" className="small text-muted d-block"> Name
+                  <input
+                    type="text"
+                    id="join-name"
+                    placeholder="Enter your name"
+                    className="form-control"
+                    onChange={(event) => setName(event.target.value)}
+                  />
+                </label>
+                { (room?.viewer_access_code || (room?.viewer_access_code && room?.moderator_access_code))
```

Comment:
I think this can just be `room?.viewer_access_code` instead of doing both checks